pkg/discovery/dtofactory/pod_entity_dto_builder.go

```diff
@@ -349,8 +357,46 @@ func (builder *podEntityDTOBuilder) getQuotaCommoditiesBought(providerUID string
 	return commoditiesBought, nil
 }
 
+// Build the CommodityDTOs bought by the pod from the Volumes.
+func (builder *podEntityDTOBuilder) buyCommoditiesFromVolumes(pod *api.Pod, mounts []repository.MountedVolume, dtoBuilder *sdkbuilder.EntityDTOBuilder) error {
+	podKey := util.PodKeyFunc(pod)
+
+	for _, mount := range mounts {
+		mountName := mount.MountName
+		volEntityID := util.PodVolumeMetricId(podKey, mountName)
+		// We use <ns>/<pod-name>/<mounted-vol-name> as the commodity key
+		// to keep the relationship between pod and volume unique.
+		commKey := fmt.Sprintf("%s/%s", podKey, mountName)
+		commBought, err := builder.getResourceCommodityBoughtWithKey(metrics.PodType, volEntityID,
+			metrics.StorageAmount, commKey, nil, nil)
+		if err != nil {
+			glog.Errorf("Failed to build %s bought by pod %s mounted %s from volume %s: %v",
+				metrics.StorageAmount, podKey, mountName, mount.UsedVolume.Name, err)
+			return err
+		}
+
+		if mount.UsedVolume == nil {
+			glog.Errorf("Error when create commoditiesBought for pod %s mounting %s: Cannot find uuid for provider "+
+				"Vol: ", podKey, mountName)
+			continue
+		}
+
+		providerVolUID := string(mount.UsedVolume.UID)
+
+		provider := sdkbuilder.CreateProvider(proto.EntityDTO_VIRTUAL_VOLUME, providerVolUID)
+		dtoBuilder = dtoBuilder.Provider(provider)
+
+		// Each pod mounts any given volume only once
+		var singleCommoditySlice []*proto.CommodityDTO
+		singleCommoditySlice = append(singleCommoditySlice, commBought)
+		dtoBuilder.BuysCommodities(singleCommoditySlice)
```

Comment:
nit: would be simpler to just use one line:
```
dtoBuilder.BuysCommodities([]*proto.CommodityDTO{commBought})
```

Reply:
Nice catch. Updated!